pkg/trainer/training.go

```diff
@@ -363,10 +362,8 @@ func (j *TrainingJob) Reconcile(config *tfv1alpha1.ControllerConfig) error {
 		return err
 	}
 
-	// TODO(jlewi): Can we determine from the CRD status whether we should
-	// Create the resources or not? We need to ensure the resources exist so for
-	// now we always call Create.
-	if j.job.Status.Phase == tfv1alpha1.TFJobPhaseCreating || j.job.Status.Phase == tfv1alpha1.TFJobPhaseRunning {
+	// Use Status.Phase to determine whether we should create the resources or not.
+	if j.job.Status.Phase == tfv1alpha1.TFJobPhaseCreating {
```

Comment:
What happens if a Job is running and one the resources like one of the services or job controllers is deleted?

Reply:
This will be handled in this PR https://github.com/kubeflow/tf-operator/pull/344

Reply:
It looks to me like #344 takes care of recreating pods if they go away but not services.
My expectation is that if a user accidentally deletes the service the job will either recover by recreating it (preferable) or fail the job.

You could follow the same pattern for services as you do for pods; i.e. do the service create in a function syncServices that gets called periodically.

Reply:
@jlewi Hi, syncServices() has been added in https://github.com/kubeflow/tf-operator/pull/344